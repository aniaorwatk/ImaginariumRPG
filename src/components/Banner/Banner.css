.banner {
  border-radius: 23px;
  margin-bottom: 24px;
  width: 342px;
  height: 100px;
}

@media only screen and (min-width: 833px) {
  .banner {
    width: 467px;
  }
}

@media only screen and (min-width: 1439px) {
  .banner {
    width: 392px;
  }
}
