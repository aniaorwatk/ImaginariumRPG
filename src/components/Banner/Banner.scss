@import "./../scss/Main";

.bannerImg {
  border-radius: 23px;
  margin-bottom: 24px;
  width: 342px;
  height: 100px;
}

@media only screen and (min-width: $breakpointM) {
  .bannerImg {
    width: 467px;
  }
}

@media only screen and (min-width: $breakpointL) {
  .bannerImg {
    width: 392px;
  }
}
