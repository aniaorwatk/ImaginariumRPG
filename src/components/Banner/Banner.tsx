import { useEffect, useState } from "react"
import { getAPI, IBannerType, IEventsType, IMediaTyp, IPatroniteType, URL_IMAGINARIUM } from "../../API/APIGet"

const Banner =()=>{

    const [banner, setBanner] = useState<IBannerType>()

    const action = (res: IMediaTyp | IBannerType | IEventsType | IPatroniteType) => {
        const response = res.story.content
        setBanner(response.banner)
      }
    
      useEffect(() => {
        getAPI(URL_IMAGINARIUM, action)
      }, []);
    return(
        <img src={banner?.filename} alt="banner" className="banner" />
    )
}

export default Banner