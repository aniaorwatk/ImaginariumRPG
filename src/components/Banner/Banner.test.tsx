import { screen, render } from "@testing-library/react";
import Banner from "./Banner";

describe('Banner', () => {

    it('Should render', async () => {
        const { asFragment } = render(<Banner />);

        await screen.findByAltText("Baner świąteczny");

        expect(asFragment()).toMatchSnapshot();
    })
})
