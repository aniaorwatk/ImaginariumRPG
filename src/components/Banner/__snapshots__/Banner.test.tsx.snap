// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Should render 1`] = `
<DocumentFragment>
  <img
    alt="Baner świąteczny"
    class="bannerImg"
    src="https://a.storyblok.com/f/188945/342x100/34919fd073/banner.jpg"
  />
</DocumentFragment>
`;
